lxd/networks.go

```diff
@@ -1564,20 +1577,28 @@ func networkStartup(s *state.State) error {
 				case <-t.C:
 					t.Stop()
 
-					// Try initializing remaining networks in random order.
 					tryInstancesStart := false
-					for pn := range initNetworks {
-						err := loadAndInitNetwork(pn.ProjectName, pn.NetworkName, false)
-						if err != nil {
-							logger.Error("Failed initializing network", logger.Ctx{"project": pn.ProjectName, "network": pn.NetworkName, "err": err})
 
-							continue
+					// Try initializing networks in priority order.
+					for priority := range initNetworks {
+						for pn := range initNetworks[priority] {
+							err := loadAndInitNetwork(pn, priority, false)
+							if err != nil {
+								logger.Error("Failed initializing network", logger.Ctx{"project": pn.ProjectName, "network": pn.NetworkName, "err": err})
+
+								continue
+							}
+
+							tryInstancesStart = true // We initialized at least one network.
 						}
+					}
 
-						tryInstancesStart = true // We initialized at least one network.
+					remainingNetworks := 0
+					for _, networks := range initNetworks {
+						remainingNetworks += len(networks)
 					}
 
-					if len(initNetworks) <= 0 {
+					if remainingNetworks <= 0 {
```

Comment:
Yeah it can't, its just a habit of defensive programming to catch all the possibilities I'm not interested in. If returned a uint I wouldn't add it. Guess I could cast it to one but seems unnecessary.

Reply:
I suppose !>1 may be clearer

Reply:
Ah ofc <1 I'll use